include/mongo_protocol.hrl

```diff
@@ -44,6 +45,25 @@
   projector = #{} :: mc_worker_api:projector()
 }).
 
+-record(op_msg_write_op, {
+  command :: command(),
+  collection :: colldb(),
+  database :: [] | mc_worker_api:database(),
```

Comment:
Maybe

```suggestion
  database = [] :: mc_worker_api:database(),
```
?

Since places that create this record without defining this field will have `undefined` instead of `[]`.

Reply:
Good suggestion. Maybe `database :: undefined | mc_worker_api:database(),` would work equally well unless we explicitly set the field to `[]` somewhere (in which case I can change those places).